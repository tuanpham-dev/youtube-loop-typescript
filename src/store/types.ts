export interface Video {
  id: number
  youtubeId: string
  volume: number | null
  range: [number, number] | [null, null]
}

export interface State {
  videos: Video[]
  playingVideo: number | null
}

export const ADD_VIDEO_BY_YOUTUBE_ID = 'ADD_VIDEO_BY_YOUTUBE_ID'
export const ADD_VIDEO = 'ADD_VIDEO'
export const REMOVE_VIDEO = 'REMOVE_VIDEO'
export const EDIT_VIDEO = 'EDIT_VIDEO'
export const PLAY_VIDEO = 'PLAY_VIDEO'
export const PLAY_FIRST_VIDEO = 'PLAY_FIRST_VIDEO'
export const PLAY_NEXT_VIDEO = 'PLAY_NEXT_VIDEO'
export const PLAY_PREVIOUS_VIDEO = 'PLAY_PREVIOUS_VIDEO'
export const UPDATE_VIDEOS = 'UPDATE_VIDEOS'

interface addVideoByYouTubeIdAction {
  type: typeof ADD_VIDEO_BY_YOUTUBE_ID
  payload: string
}

interface addVideoAction {
  type: typeof ADD_VIDEO
  payload: Video
}

interface removeVideoAction {
  type: typeof REMOVE_VIDEO
  payload: number
}

interface editVideoAction {
  type: typeof EDIT_VIDEO
  payload: Video
}

interface updateVideosAction {
  type: typeof UPDATE_VIDEOS
  payload: Video[]
}

interface playVideoAction {
  type: typeof PLAY_VIDEO
  payload: number
}

interface playFirstVideoAction {
  type: typeof PLAY_FIRST_VIDEO
}

interface playPreviousVideoAction {
  type: typeof PLAY_PREVIOUS_VIDEO
}

interface playNextVideoAction {
  type: typeof PLAY_NEXT_VIDEO
}

interface stopVideoAction {
  type: typeof PLAY_VIDEO
  payload: null
}

type ActionByType<T extends ActionType['type'], A = ActionType> = A extends {type: T} ? A : never

export type ActionType = addVideoByYouTubeIdAction | addVideoAction | removeVideoAction
  | editVideoAction| updateVideosAction | playVideoAction | playFirstVideoAction
  | playPreviousVideoAction | playNextVideoAction | stopVideoAction  
export type ReducerHandler<T extends ActionType['type']> = (state: State, action: ActionByType<T>) => State
export type ReducerMap = {
  [T in ActionType['type']]?: ReducerHandler<T>
}