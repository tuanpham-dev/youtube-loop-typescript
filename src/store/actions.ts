import {
  Video,
  ActionType,
  ADD_VIDEO_BY_YOUTUBE_ID,
  ADD_VIDEO, REMOVE_VIDEO,
  EDIT_VIDEO,
  UPDATE_VIDEOS,
  PLAY_VIDEO,
  PLAY_PREVIOUS_VIDEO,
  PLAY_NEXT_VIDEO,
  PLAY_FIRST_VIDEO
} from './types'
import getYouTubeVideoId from '../utils/getYouTubeVideoId'

export const addVideoByYouTubeId = (youtubeId: string): ActionType => {
  const youtubeVideoId = getYouTubeVideoId(youtubeId)

  return {
    type: ADD_VIDEO_BY_YOUTUBE_ID,
    payload: youtubeVideoId
  }
}

export const addVideo = (video: Video): ActionType => {
  return {
    type: ADD_VIDEO,
    payload: video
  }
}

export const removeVideo = (videoId: number): ActionType => {
  return {
    type: REMOVE_VIDEO,
    payload: videoId
  }
}

export const editVideo = (video: Video): ActionType => {
  return {
    type: EDIT_VIDEO,
    payload: video
  }
}

export const updateVideos = (videos: Video[]): ActionType => {
  return {
    type: UPDATE_VIDEOS,
    payload: videos
  }
}

export const playVideo = (videoId: number): ActionType => {
  return {
    type: PLAY_VIDEO,
    payload: videoId
  }
}

export const playFirstVideo = (): ActionType => {
  return {
    type: PLAY_FIRST_VIDEO
  }
}

export const playPreviousVideo = (): ActionType => {
  return {
    type: PLAY_PREVIOUS_VIDEO
  }
}

export const playNextVideo = (): ActionType => {
  return {
    type: PLAY_NEXT_VIDEO
  }
}

export const stopVideo = (): ActionType => {
  return {
    type: PLAY_VIDEO,
    payload: null
  }
}
