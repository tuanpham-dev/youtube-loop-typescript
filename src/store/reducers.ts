import {
  Video,
  State,
  ActionType,
  ReducerMap,
  ADD_VIDEO_BY_YOUTUBE_ID,
  ADD_VIDEO,
  REMOVE_VIDEO,
  EDIT_VIDEO,
  UPDATE_VIDEOS,
  PLAY_VIDEO,
  PLAY_FIRST_VIDEO,
  PLAY_PREVIOUS_VIDEO,
  PLAY_NEXT_VIDEO,
  ReducerHandler
} from "./types";

const initialState: State = {
  videos: [],
  playingVideo: null
}

const reducerMap: ReducerMap = {
  [ADD_VIDEO_BY_YOUTUBE_ID]: (state, { payload: youtubeId }) => {
    const maxId = Math.max(...state.videos.map(video => video.id), 0)
    const video: Video = {
      id: maxId + 1,
      youtubeId,
      volume: null,
      range: [null, null]
    }

    return {
      ...state,
      videos: [...state.videos, video]
    }
  },

  [ADD_VIDEO]: (state, { payload: video }) => {
    const maxId = Math.max(...state.videos.map(video => video.id), 0)

    return {
      ...state,
      videos: [...state.videos, { ...video, id: maxId + 1 }]
    }
  },

  [REMOVE_VIDEO]: (state, { payload: videoId }) => {
    const index = state.videos.findIndex(video => video.id === videoId)
    let playingVideo: number | null = null

    if (index !== null) {
      if (state.playingVideo === videoId) {
        if (state.videos.length <= 1) {
          playingVideo = null
        } else {
          playingVideo = state.videos[index === state.videos.length - 1 ? 0 : index + 1].id
        }
      }

      const videos = [...state.videos]
      videos.splice(index, 1)

      if (playingVideo !== null) {
        return {
          ...state,
          videos,
          playingVideo
        }
      } else {
        return {
          ...state,
          videos
        }
      }
    }

    return state
  },

  [EDIT_VIDEO]: (state, { payload: video }) => {
    const index = state.videos.findIndex(videoItem => videoItem.id === video.id)

    if (index !== null) {
      const videos = [...state.videos]
      videos.splice(index, 1, video)

      return {
        ...state,
        videos
      }
    }

    return state
  },

  [UPDATE_VIDEOS]: (state, { payload: videos }) => {
    return {
      ...state,
      videos
    }
  },

  [PLAY_VIDEO]: (state, { payload: videoId }) => {
    if (state.playingVideo !== videoId) {
      return {
        ...state,
        playingVideo: videoId
      }
    }

    return state
  },

  [PLAY_FIRST_VIDEO]: (state) => {
    if (state.videos.length > 0) {
      return {
        ...state,
        playingVideo: state.videos[0].id
      }
    }

    return state
  },

  [PLAY_PREVIOUS_VIDEO]: (state) => {
    if (state.videos.length > 0) {
      let previousVideoIndex
      const index = state.videos.findIndex(video => video.id === state.playingVideo)
  
      if (index === 0) {
        previousVideoIndex = state.videos.length - 1
      } else {
        previousVideoIndex = index -1
      }

      return {
        ...state,
        playingVideo: state.videos[previousVideoIndex].id
      }
    }

    return state
  },

  [PLAY_NEXT_VIDEO]: (state) => {
    if (state.videos.length > 0) {
      let nextVideoIndex
      const index = state.videos.findIndex(video => video.id === state.playingVideo)

      if (index === state.videos.length - 1) {
        nextVideoIndex = 0
      } else {
        nextVideoIndex = index + 1
      }

      return {
        ...state,
        playingVideo: state.videos[nextVideoIndex].id
      }
    }

    return state
  }
}

export const rootReducer = (state = initialState, action: ActionType): State => {
  if (typeof reducerMap[action.type] === 'function') {
    const reducerHandler = reducerMap[action.type] as ReducerHandler<typeof action.type>

    return reducerHandler(state, action)
  } else {
    return state
  }
}

export type RootState = ReturnType<typeof rootReducer>
export default rootReducer