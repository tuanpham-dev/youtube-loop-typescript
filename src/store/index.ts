import { Store, createStore } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import { rootReducer, RootState } from './reducers'

export const configureStore = (initialState?: RootState): Store<RootState> => {
  const middleware = composeWithDevTools()
  const store = createStore(rootReducer, initialState, middleware) as Store<RootState>

  return store
}

export default configureStore