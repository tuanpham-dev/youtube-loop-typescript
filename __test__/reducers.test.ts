import rootReducer from '../src/store/reducers'
import mockData from './mock-data/video'
import { ADD_VIDEO_BY_YOUTUBE_ID, ADD_VIDEO, REMOVE_VIDEO, EDIT_VIDEO, UPDATE_VIDEOS, PLAY_VIDEO, PLAY_FIRST_VIDEO, PLAY_PREVIOUS_VIDEO, PLAY_NEXT_VIDEO } from '../src/store/types'

const initialState = {
  videos: [],
  playingVideo: null
}

describe('root reducer', () => {
  it('should return the initial state', () => {
    const reducer = rootReducer(undefined, {} as any)

    expect(reducer).toEqual(initialState)
    expect(reducer).toMatchSnapshot()
  })

  it('should handle ADD_VIDEO_BY_YOUTUBE_ID', () => {
    expect(rootReducer(initialState, {
      type: ADD_VIDEO_BY_YOUTUBE_ID,
      payload: mockData.youtubeId
    })).toMatchSnapshot()

    expect(rootReducer({
      ...initialState,
      videos: [mockData.video]
    }, {
      type: ADD_VIDEO_BY_YOUTUBE_ID,
      payload: mockData.youtubeId
    })).toMatchSnapshot()
  })

  it('should handle ADD_VIDEO', () => {
    expect(rootReducer(initialState, {
      type: ADD_VIDEO,
      payload: mockData.video
    })).toMatchSnapshot()

    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: ADD_VIDEO,
      payload: mockData.video
    })).toMatchSnapshot()
  })

  it('should handle REMOVE_VIDEO', () => {
    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: REMOVE_VIDEO,
      payload: mockData.videos[1].id
    })).toMatchSnapshot()

    expect(rootReducer({
      videos: mockData.videos,
      playingVideo: mockData.videos[1].id
    }, {
      type: REMOVE_VIDEO,
      payload: mockData.videos[1].id
    })).toMatchSnapshot()

    expect(rootReducer({
      videos: mockData.videos,
      playingVideo: mockData.videos[1].id
    }, {
      type: REMOVE_VIDEO,
      payload: 123
    })).toMatchSnapshot()
  })

  it('should handle EDIT_VIDEO', () => {
    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: EDIT_VIDEO,
      payload: mockData.video
    })).toMatchSnapshot()

    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: EDIT_VIDEO,
      payload: {
        id: 123,
        youtubeId: '',
        volume: null,
        range: [null, null]
      }
    })).toMatchSnapshot()
  })

  it('should handle UPDATE_VIDEOS', () => {
    expect(rootReducer({
      ...initialState
    }, {
      type: UPDATE_VIDEOS,
      payload: mockData.videos
    })).toMatchSnapshot()

    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: UPDATE_VIDEOS,
      payload: []
    })).toMatchSnapshot()
  })

  it('should handle PLAY_VIDEO', () => {
    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: PLAY_VIDEO,
      payload: mockData.videos[2].id
    })).toMatchSnapshot()
  })

  it('should handle PLAY_FIRST_VIDEO', () => {
    expect(rootReducer({
      ...initialState,
      videos: mockData.videos
    }, {
      type: PLAY_FIRST_VIDEO
    })).toMatchSnapshot()
  })

  it('should handle PLAY_PREVIOUS_VIDEO', () => {
    expect(rootReducer({
      videos: mockData.videos,
      playingVideo: mockData.videos[2].id
    }, {
      type: PLAY_PREVIOUS_VIDEO
    })).toMatchSnapshot()

    expect(rootReducer({
      videos: mockData.videos,
      playingVideo: mockData.videos[0].id
    }, {
      type: PLAY_PREVIOUS_VIDEO
    })).toMatchSnapshot()
  })

  it('should handle PLAY_NEXT_VIDEO', () => {
    expect(rootReducer({
      videos: mockData.videos,
      playingVideo: mockData.videos[2].id
    }, {
      type: PLAY_NEXT_VIDEO
    })).toMatchSnapshot()

    expect(rootReducer({
      videos: mockData.videos,
      playingVideo: mockData.videos[mockData.videos.length - 1].id
    }, {
      type: PLAY_NEXT_VIDEO
    })).toMatchSnapshot()
  })
})