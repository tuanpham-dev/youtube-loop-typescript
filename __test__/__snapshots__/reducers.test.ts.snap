// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root reducer should handle ADD_VIDEO 1`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        230,
      ],
      "volume": 50,
      "youtubeId": "nfWlot6h_JM",
    },
  ],
}
`;

exports[`root reducer should handle ADD_VIDEO 2`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
    Object {
      "id": 5,
      "range": Array [
        0,
        230,
      ],
      "volume": 50,
      "youtubeId": "nfWlot6h_JM",
    },
  ],
}
`;

exports[`root reducer should handle ADD_VIDEO_BY_YOUTUBE_ID 1`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        null,
        null,
      ],
      "volume": null,
      "youtubeId": "nfWlot6h_JM",
    },
  ],
}
`;

exports[`root reducer should handle ADD_VIDEO_BY_YOUTUBE_ID 2`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        230,
      ],
      "volume": 50,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        null,
        null,
      ],
      "volume": null,
      "youtubeId": "nfWlot6h_JM",
    },
  ],
}
`;

exports[`root reducer should handle EDIT_VIDEO 1`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        230,
      ],
      "volume": 50,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle EDIT_VIDEO 2`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 123,
      "range": Array [
        null,
        null,
      ],
      "volume": null,
      "youtubeId": "",
    },
  ],
}
`;

exports[`root reducer should handle PLAY_FIRST_VIDEO 1`] = `
Object {
  "playingVideo": 1,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle PLAY_NEXT_VIDEO 1`] = `
Object {
  "playingVideo": 4,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle PLAY_NEXT_VIDEO 2`] = `
Object {
  "playingVideo": 1,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle PLAY_PREVIOUS_VIDEO 1`] = `
Object {
  "playingVideo": 2,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle PLAY_PREVIOUS_VIDEO 2`] = `
Object {
  "playingVideo": 4,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle PLAY_VIDEO 1`] = `
Object {
  "playingVideo": 3,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle REMOVE_VIDEO 1`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle REMOVE_VIDEO 2`] = `
Object {
  "playingVideo": 3,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle REMOVE_VIDEO 3`] = `
Object {
  "playingVideo": 2,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
  ],
}
`;

exports[`root reducer should handle UPDATE_VIDEOS 1`] = `
Object {
  "playingVideo": null,
  "videos": Array [
    Object {
      "id": 1,
      "range": Array [
        0,
        242,
      ],
      "volume": 100,
      "youtubeId": "nfWlot6h_JM",
    },
    Object {
      "id": 2,
      "range": Array [
        0,
        256,
      ],
      "volume": 100,
      "youtubeId": "3tmd-ClpJxA",
    },
    Object {
      "id": 3,
      "range": Array [
        0,
        273,
      ],
      "volume": 100,
      "youtubeId": "e-ORhEE9VVg",
    },
    Object {
      "id": 4,
      "range": Array [
        0,
        245,
      ],
      "volume": 100,
      "youtubeId": "QcIy9NiNbmo",
    },
  ],
}
`;

exports[`root reducer should handle UPDATE_VIDEOS 2`] = `
Object {
  "playingVideo": null,
  "videos": Array [],
}
`;

exports[`root reducer should return the initial state 1`] = `
Object {
  "playingVideo": null,
  "videos": Array [],
}
`;
